core/src/main/java/org/apache/accumulo/core/gc/ReferenceFile.java

```diff
@@ -30,6 +31,7 @@ public class ReferenceFile implements Reference, Comparable<ReferenceFile> {
   // parts of an absolute URI, like "hdfs://1.2.3.4/accumulo/tables/2a/t-0003"
   public final TableId tableId; // 2a
   public final boolean isScan;
+  public boolean isDirectory = false;
```

Comment:
ReferenceFile types shouldn't change so this boolean should be final
```suggestion
  public final boolean isDirectory;
```

This could be supported by adding another boolean to the protected constructor's method signature. 
```
  protected ReferenceFile(TableId tableId, String metadataEntry, boolean isScan, boolean isDirectory) {
    this.tableId = Objects.requireNonNull(tableId);
    this.metadataEntry = Objects.requireNonNull(metadataEntry);
    this.isScan = isScan;
    this.isDirectory = isDirectory
  }
```

And modifying the `forDirectory` constructor

```
  public static ReferenceFile forDirectory(TableId tableId, String dirName) {
    MetadataSchema.TabletsSection.ServerColumnFamily.validateDirCol(dirName);
    return new ReferenceFile(tableId, dirName, false, true);
  }
```

Reply:
Done, updated.